### ISSUE TITLE
`invalid-type-form` false positive on paramspec when used in the return type

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3316]
### `main.py`

```py
from typing import Callable, cast


def fn[**P]() -> Callable[P, None]: # removing the return type annotation makes the errors go away
    a: Callable[P, None] = cast(Callable[P, None], object())
    return cast(Callable[P, None], object())
```

```
The first argument to `Callable` must be either a list of types, ParamSpec, Concatenate, or `...` (invalid-type-form) [Ln 5, Col 17]
The first argument to `Callable` must be either a list of types, ParamSpec, Concatenate, or `...` (invalid-type-form) [Ln 6, Col 26]
```

### Version

ty 0.0.32 (4d1e1fc57 2026-04-20)

**Comment by dhruvmanila** [comment]
This is probably related to https://github.com/astral-sh/ruff/pull/22954

**Comment by carljm** [comment]
Can you show a more complete code example where this arose in real code? I'm just curious about the use case, since it seems like `P` is useless here; the function could just as well return `Callable[..., None]` with no observable difference in behavior.

**Comment by DetachHead** [comment]
it seems to come up on any functions that return decorators where you want to use the paramspec in a type annotation before returning the wrapped function. i don't have an exact real life example because it's a private codebase, but something like this is a bit more realistic i guess:

```py
from typing import Callable


def log_on_call[**P, R](message: str | None) -> Callable[[Callable[P, R]], Callable[P, R]]:

    def deco(fn: Callable[P, R]) -> Callable[P, R]:
        result: Callable[P, R] # error
        if message:

            def wrapped(*args: P.args, **kwargs: P.kwargs) -> R:
                print(message)
                return fn(*args, **kwargs)

            result = wrapped
        else:
            # don't wrap the function unless it's necessary
            result = fn
        return result

    return deco
```

(note that this example is full of other false positive errors which i believe are caused by #3317)
